pkg/koordlet/metricsadvisor/collectors/performance/performance_collector_linux.go

```diff
@@ -99,59 +100,50 @@ func (p *performanceCollector) collectContainerCPI() {
 			containerStatusesMap[containerStat] = meta
 		}
 	}
-	// get container CPI collectors for each container
-	collectors := sync.Map{}
+	metricsChan := make(chan metriccache.MetricSample, len(containerStatusesMap)*2)
+	pciMetrics := make([]metriccache.MetricSample, 0)
+	// get container PCI Metric
+	go p.getContainerPCIMetrics(containerStatusesMap, metricsChan)
```

Comment:
go p.getContainerPCIMetrics() -> p.getContainerPCIMetrics()
should save metrics only if all metrics have been coillected.